src/m365/spo/commands/tenant/tenant-homesite-add.spec.ts

```diff
@@ -0,0 +1,175 @@
+    "Title": "testcommsite",
+    "Url": homeSite,
+    "VivaConnectionsDefaultStart": false,
+    "WebId": "256c4f0f-e372-47b4-a891-b4888e829e20"
+  };
+
+  const homeSiteConfig = {
+    "Audiences": [
+      {
+        "Email": "SharingTest@reshmeeauckloo.onmicrosoft.com",
+        "Id": "af8c0bc8-7b1b-44b4-b087-ffcc8df70d16",
+        "Title": "SharingTest Members"
+      }
+    ],
+    "IsInDraftMode": true,
+    "IsVivaBackendSite": false,
+    "SiteId": "ca49054c-85f3-41eb-a290-46ffda8f219c",
+    "TargetedLicenseType": 0,
+    "Title": "testcommsite",
+    "Url": "https://contoso.sharepoint.com/sites/testcomms",
+    "VivaConnectionsDefaultStart": false,
+    "WebId": "256c4f0f-e372-47b4-a891-b4888e829e20"
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.connection.active = true;
+    auth.connection.spoUrl = 'https://contoso.sharepoint.com';
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+    auth.connection.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_HOMESITE_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds available home sites', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-admin.sharepoint.com/_api/SPHSite/AddHomeSite`) {
+        return homeSites;
+      }
+
+      throw opts.url;
+    });
+
+    await command.action(logger, { options: { url: homeSite, verbose: true } });
+    assert(loggerLogSpy.calledWith(homeSites));
+  });
+
+  it('adds a home site with the specified URL, isInDraftMode, vivaConnectionsDefaultStart, and audiences', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-admin.sharepoint.com/_api/SPHSite/AddHomeSite`) {
+        return homeSiteConfig;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        url: homeSite,
+        isInDraftMode: 'true',
+        vivaConnectionsDefaultStart: 'false',
+        audiences: 'af8c0bc8-7b1b-44b4-b087-ffcc8df70d16,754ff15c-76b1-44cb-88c7-0065a4d3cfb7',
+        order: 2
+      }
+    });
+    assert(loggerLogSpy.calledWith(homeSiteConfig));
```

Comment:
Instead of checking just this, let's check if we actually provided the right arguments in our request to SharePoint. An example can be found here:

https://github.com/pnp/cli-microsoft365/blob/db90bedc910c97189a79022efab478379ae386cc/src/m365/spo/commands/site/site-admin-add.spec.ts#L314-L320